### PR TITLE
Worklogs extended support

### PR DESCRIPTION
**Pull request by falnyr**
Hi,
I'm using your package as a dependency in my project. Since there is a couple of features missing I'd like to contribute a little bit. To be able to override the dependency on your package with my fork I had to migrate the project to `go modules`. 

With these changes, is it okay to merge to master or you would prefer a new version?

Thanks

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+go 1.12
+
+require (
+	github.com/fatih/structs v1.0.0
+	github.com/google/go-querystring v0.0.0-20170111101155-53e6ce116135
+	github.com/pkg/errors v0.8.0
+	github.com/trivago/tgo v1.0.1
+	golang.org/x/crypto v0.0.0-20190426145343-a29dc8fdc734
+)
```
Comment (ghostsquad) [inline go.mod]: Please also update the readme to indicate changes to the development workflow

File: issue_test.go [hunk 2]
```diff
@@ -262,7 +262,7 @@ func TestIssueService_AddWorklogRecord(t *testing.T) {
 	r := &WorklogRecord{
 		TimeSpent: "1h",
 	}
-	record, _, err := testClient.Issue.AddWorklogRecord("10000", r)
```
[the remark this answers is not in the document]
Comment (falnyr) [reply issue_test.go]: Thanks, I will look into it!

[the remark this answers is not in the document]
Comment (ghostsquad) [reply issue_test.go]: Great! Looking forward to it.

[the remark this answers is not in the document]
Comment (falnyr) [reply issue_test.go]: Would this make more sense now?

**Comment by ghostsquad** [comment]
The tests seem to have disappeared. Please ensure you have positive and negative tests

File: issue_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ghostsquad) [reply issue_test.go]: Sorry for the delay in getting back to this PR. I'm looking at it now.

**Comment by ghostsquad** [comment]
@falnyr this looks good! The tests look like they are failing. I'll evaluate when you get the tests passing. Thanks for your contributions!

**Comment by falnyr** [comment]
I am still working on more complex testing scenarios. I'd like to compare the results of service methods with "expected" struct. I'm using `go-cmp` for that but there are some slight hiccups with the `time` package for `Created, Updated, Started` fields. I'm hoping to resolve these within a few days.

**Comment by ghostsquad** [comment]
If you need a way to control time, may I recommend a package I created?

https://github.com/ghostsquad/go-timejumper/

Let's you freeze time, and make tests deterministic. I'm not sure if that's the problem you are running into though.

**Comment by ghostsquad** [comment]
@falnyr - I deprecated go 1.7 which fixed the tests. Is this ready for a rebase & squash?

**Comment by falnyr** [comment]
@ghostsquad added another test. I think we should be fine to merge now if you're happy with it, I'll be adding more changes in the next PRs.

**Comment by ghostsquad** [comment]
@falnyr if you don't mind, can you rebase and squash your commits, and follow the following standard proposal (below) for commit messages? I'm playing around with making release notes automated.

https://www.conventionalcommits.org/en/v1.0.0-beta.3/

**Comment by falnyr** [comment]
@ghostsquad Would this be alright?

**Pull request merged**